src/loader.c

```diff
@@ -547,6 +547,7 @@ bool parse_pool(void *base, const char *name, const char *params)
 		goto fail;
 	}
 	pool->pool_size = pool_size;
+	notify_pool_event(pool, handle_pool_cf_update);
```

Comment:
We activate a callback (queued) that will enforce pool's max connection limit and evict any connections above newly configured limit.